tests/test_dataset_mask.py

```diff
@@ -176,6 +191,20 @@ def test_kwargs(tiffs):
         other = src.dataset_mask(window=((1, 4), (1, 4)), boundless=True)
         assert np.array_equal(alp_shift_lr, other)
 
+        other = src.dataset_mask(out_shape=(1, 5, 5))
+        assert np.array_equal(resampmask, other)
+
+        out = np.zeros((1, 5, 5), dtype=np.uint8)
+        other = src.dataset_mask(out=out)
+        assert np.array_equal(resampmask, other)
+
         # band indexes are not supported
         with pytest.raises(TypeError):
             src.dataset_mask(indexes=1)
+
+
+def test_kwargs_resampling(tiffs):
```

Comment:
@vincentsarago add `@requires_gdal2(reason="GDAL 2+ required for resampling")` before this function and we'll skip the test with GDAL 1.11.

This should be it!

Reply:
🤦‍♂️